interfaces/tests/file_int/file_int.c

```diff
@@ -0,0 +1,559 @@
+
+    ///act
+    FILE_WRITE_ASYNC_RESULT write_result = file_write_async(file_handle, source, size, 0, write_callback, &write_context);
+    wait_on_address_helper(&write_context.value, &write_context.pre_callback_value, UINT32_MAX);
+    
+    ///assert
+    ASSERT_ARE_EQUAL(FILE_WRITE_ASYNC_RESULT, FILE_WRITE_ASYNC_OK, write_result);
+    ASSERT_ARE_EQUAL(int32_t, write_context.post_callback_value, interlocked_or(&write_context.value, 0), "value should be post_callback_value");
+    ASSERT_IS_TRUE(write_context.did_write_succeed);
+
+    ///act
+    file_read_async(file_handle, destination, sizeof(destination), read_position, read_callback, &read_context);
+    wait_on_address_helper(&read_context.value, &read_context.pre_callback_value, UINT32_MAX);
+
+    ///assert
+    ASSERT_ARE_EQUAL(int32_t, read_context.post_callback_value, interlocked_or(&read_context.value, 0), "value should be post_callback_value");
+    ASSERT_IS_FALSE(read_context.did_read_succeed);
+
+    //cleanup
+    file_destroy(file_handle);
+}
+
+/*Tests_SRS_FILE_43_003: [If a file with name full_file_name does not exist, file_create shall create a file with that name.]*/
+/*Tests_SRS_FILE_43_001: [file_create shall open the file named full_file_name for asynchronous operations and return its handle.]*/
+/*Tests_SRS_FILE_43_006: [file_destroy shall wait for all pending I / O operations to complete.]*/
+/*Tests_SRS_FILE_43_007: [file_destroy shall close the file handle handle.]*/
+/*Tests_SRS_FILE_43_014: [file_write_async shall enqueue a write request to write source's content to the position offset in the file. ]*/
+/*Tests_SRS_FILE_43_041: If position + size is greater than the size of the file and the call to write is successfull, file_write_async shall grow the file to accomodate the write.]*/
+/*Tests_SRS_FILE_43_008: [file_write_async shall call user_call_back passing user_context and success depending on the success of the asynchronous write operation.]*/
+/*Tests_SRS_FILE_43_030: [file_write_async shall succeed and return FILE_WRITE_ASYNC_OK.]*/
+/*Tests_SRS_FILE_43_021: [file_read_async shall enqueue a read request to read handle's content at position offset and write it to destination. ]*/
+/*Tests_SRS_FILE_43_016: [file_read_async shall call user_callback passing user_context and success depending on the success of the asynchronous read operation.]*/
+/*Tests_SRS_FILE_43_031: [file_read_async shall succeed and return FILE_READ_ASYNC_OK.]*/
+TEST_FUNCTION(large_simultaneous_writes_succeed)
+{
+    ///arrange
+    int block_size = 4096;
+    int num_blocks = 50;
+    WRITE_COMPLETE_CONTEXT contexts[50];
+    unsigned char* sources[50];
+
+    char filename[] = "large_simultaneous_writes_succeed.txt";
+    ASSERT_IS_FALSE(check_file_exists(filename));
+    EXECUTION_ENGINE_HANDLE execution_engine = execution_engine_create(NULL);
+    FILE_HANDLE file_handle = file_create(execution_engine, filename, NULL, NULL);

```

Comment:
If it's easier, you could even have a helper function "test_file_create_new_file(filename)" which would delete the file if it exists and return the file handle. Cuts down on lines of code per test.

Reply:
...because my next comment is, you are missing the ASSERT_IS_NOT_NULL(file_handle) here and everywhere

---
In reply to: [458314695](https://github.com/Azure/azure-c-pal/pull/10#discussion_r458314695) [](ancestors = 458314695)